src/sentry/static/sentry/app/views/performance/charts/footer.tsx

```diff
@@ -1,20 +1,93 @@
 import React from 'react';
+import * as Sentry from '@sentry/browser';
+import {Location} from 'history';
 
 import {t} from 'app/locale';
+import {Client} from 'app/api';
+import {fetchTotalCount} from 'app/views/eventsV2/utils';
+import EventView, {isAPIPayloadSimilar} from 'app/views/eventsV2/eventView';
+import {Organization} from 'app/types';
 
-import {ChartControls, SectionHeading} from './styles';
+import {ChartControls, SectionHeading, SectionValue} from './styles';
 
 type Props = {
-  totals: number | null;
+  api: Client;
+  eventView: EventView;
+  organization: Organization;
+  location: Location;
 };
 
-export default function ChartFooter({totals}: Props) {
-  return (
-    <ChartControls>
-      <SectionHeading>
-        {t('Total Events')}
-        {totals}
-      </SectionHeading>
-    </ChartControls>
-  );
+type State = {
+  totalValues: null | number;
+};
+
+class ChartFooter extends React.Component<Props, State> {
+  state: State = {
+    totalValues: null,
+  };
+
+  componentDidMount() {
+    this.mounted = true;
+
+    this.fetchTotalCount();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const orgSlugHasChanged =
+      this.props.organization.slug !== prevProps.organization.slug;
+    const shouldRefetch = this.shouldRefetchData(prevProps);
+
+    if ((orgSlugHasChanged || shouldRefetch) && this.props.eventView.isValid()) {
+      this.fetchTotalCount();
+    }
+  }
+
+  componentWillUnmount() {
+    this.mounted = false;
+  }
+
+  shouldRefetchData = (prevProps: Props): boolean => {
+    const thisAPIPayload = this.props.eventView.getEventsAPIPayload(this.props.location);
+    const otherAPIPayload = prevProps.eventView.getEventsAPIPayload(prevProps.location);
+
+    return !isAPIPayloadSimilar(thisAPIPayload, otherAPIPayload);
+  };
+
+  mounted: boolean = false;
+
+  async fetchTotalCount() {
+    const {api, organization, location, eventView} = this.props;
+    if (!eventView.isValid() || !this.mounted) {
+      return;
+    }
+
+    try {
+      const totals = await fetchTotalCount(
+        api,
+        organization.slug,
+        eventView.getEventsAPIPayload(location)
```

Comment:
This will include a bunch of parameters the endpoint doesn't care about. Should we add another method to eventView to get properties that aren't the fields?